### PR TITLE
test: add tests for conditional formatting

### PR DESCRIPTION
**Pull request by DiegoCardoso**
## Description

Add some JUnit tests for conditional formatting to check if cells matching formulas receive the styles.

### CODE REVIEW

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java [hunk 1]
```diff
@@ -1,11 +1,20 @@
 package com.vaadin.flow.component.spreadsheet.tests;
 
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.core.JsonParser;
+import com.nimbusds.jose.shaded.json.parser.JSONParser;
```
Comment (tomivirkki) [inline vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java:5]: These (and  a couple more) are unused imports

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java [hunk 2]
```diff
@@ -43,13 +115,15 @@ private void createConditionalFormatterRulesForSheet(String fileName) {
...
-    private void createConditionalFormatterRulesForSheet(String fileName,
+    private Spreadsheet createConditionalFormatterRulesForSheet(String fileName,
             Integer sheetIndex) {
         Spreadsheet sheet = TestHelper.createSpreadsheet(fileName);
 
         if (sheetIndex != null && sheet.getActiveSheetIndex() != sheetIndex) {
             sheet.setActiveSheetIndex(sheetIndex);
         }
-        new ConditionalFormatter(sheet).createConditionalFormatterRules();
+        // new ConditionalFormatter(sheet).createConditionalFormatterRules();
```
Comment (tomivirkki) [inline vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java:125]: Should this just be removed altogether?

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java [hunk 3]
```diff
@@ -29,10 +38,73 @@ public void createConditionalFormatterRules_sheetWithStringFormatRuleForNumericC
     @Test
     public void matchesFormula_rulesWithoutFormula_formulasEvaluatedWithoutExceptions() {
         // ensure sheet with rules without formulas is active
-        createConditionalFormatterRulesForSheet(
+        Spreadsheet sheet = createConditionalFormatterRulesForSheet(
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java]: It was a leftover of a WIP. Removed the variable.

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java (hunk 1 above)
Comment (DiegoCardoso) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java]: Removed.

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java (hunk 2 above)
Comment (DiegoCardoso) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/ConditionalFormatterTest.java]: Removed.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**